### PR TITLE
Console UI improvment

### PR DESCRIPTION
**Pull request by EvgenyAlterman**

### CODE REVIEW

File: quix-frontend/client/src/components/note/note-testkit.ts [hunk 1]
```diff
@@ -20,6 +25,6 @@ export class NoteTestkit extends Testkit {
   }
 
   async isNameEditable() {
-    return this.evaluate.hook(Hooks.Name, el => el.getAttribute('contenteditable') === 'true');
+    return this.evaluate.hook( Hooks.Name, el => el.getAttribute('contenteditable') === 'true');
```
Comment (antonpodolsky) [inline quix-frontend/client/src/components/note/note-testkit.ts:28]: extra space

File: quix-frontend/client/src/lib/runner/directives/results/console/console-result.html [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
-    'bi-white': row.level === 'INFO',
-    'bi-danger': row.level === 'ERROR'
-  }" ng-repeat="row in query.getResults().buffer"><span class="bi-white">[{{::row.timestamp}}]</span> {{::row.line}}</pre>
-</div>
\ No newline at end of file
+<div class="bi-c bi-theme--darker" data-hook="console-result"> 
+  <div 
+    ng-if="!$last && isTimestampVisible(row.timestamp, $index, $first, $last)"
+    ng-repeat-start="row in query.getResults().buffer"
+    class="cr-timestamp bi-s-v--x2"
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/runner/directives/results/console/console-result.html:5]: there seems to be some inconsistency in the class prefix, `cr` vs `console-result`

File: quix-frontend/client/src/lib/runner/directives/results/console/console-result.scss [hunk 3]
```diff
@@ -9,10 +9,72 @@ bi-console-result {
...
+      color: lighten($darker-bg, 80);
+    }
   }
-}
\ No newline at end of file
+
+  bi-copy-to-clipboard {
+    position: absolute;
+    right: 0;
+    top: 15;
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/runner/directives/results/console/console-result.scss:26]: 15 => 15px

File: quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts [hunk 4]
```diff
@@ -6,7 +7,27 @@ export default () => {
...
+        
+        scope.isTimestampVisible = (timestamp, index, isFirst, isLast) => {
+          return (
+            isFirst ||
+            isLast ||
+            scope.query.getResults().buffer[index - 1].timestamp !== timestamp
+          );
+        };
+
+        scope.groupText = (timestamp) => {
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts:23]: groupText => getTextByTimestamp

File: quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts [hunk 5]
```diff
@@ -6,7 +7,27 @@ export default () => {
     restrict: 'E',
     template,
     scope: {
-      query: '<'
-    }
+      query: '<',
+    },
+    link: {
+      async pre(scope: IScope) {
+
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts:14]: not sure if newline is justified.

**Review by antonpodolsky — CHANGES_REQUESTED** [review]
Looks good, just small stuff.

File: quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts (hunk 5 above)
Comment (EvgenyAlterman) [reply quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts:14]: Initially it made by prettier, bu in this specific case "link" is an object and expanded object is naturally looks beter with this new line in my opinion :)

Comment (EvgenyAlterman) [reply quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts:14]: but if you are about the line 14 so yes, I removed it

Comment (antonpodolsky) [reply quix-frontend/client/src/lib/runner/directives/results/console/console-result.ts:14]: yes, line 14

**Pull request merged**